### PR TITLE
extend team audit to the hidden chain

### PR DESCRIPTION
**Pull request by AMarcedone**
this depends on the pr tightening the auditor team checks. I am separating the two to make reviewing easier.

### CODE REVIEW

File: go/teams/audit.go [hunk 1]
```diff
@@ -322,8 +337,38 @@ func (a *Auditor) doPostProbes(m libkb.MetaContext, history *keybase1.AuditHisto
 				return 0, keybase1.Seqno(0), NewAuditError("team chain linkID mismatch at %d: wanted %s but got %s via merkle seqno %d", tuple.team, expectedLinkID, tuple.linkID, tuple.merkle)
 			}
 		}
+		if auditMode != keybase1.AuditMode_STANDARD_NO_HIDDEN {
```
Comment (maxtaco) [inline go/teams/audit.go]: separate function please

File: go/teams/audit.go [hunk 2]
```diff
@@ -360,8 +410,14 @@ func (a *Auditor) doPreProbes(m libkb.MetaContext, history *keybase1.AuditHistor
...
 		if tuple.team != keybase1.Seqno(0) || !tuple.linkID.IsNil() {
-			return 0, NewAuditError("merkle root at %v should have been nil for %v; got %s/%d",
-				tuple.merkle, history.ID, tuple.linkID, tuple.team)
+			return 0, NewAuditError("merkle root should not have had a leaf for team %v: got %s/%d at merkle seqno %v",
+				history.ID, tuple.linkID, tuple.team, tuple.merkle)
+		}
+		if tuple.hiddenResp.RespType == libkb.MerkleHiddenResponseTypeNONE && tuple.merkle > firstRootWithHidden && auditMode != keybase1.AuditMode_STANDARD_NO_HIDDEN {
+			return 0, NewAuditError("did not get a hidden response but one was expected (at main seqno %v)", tuple.merkle)
+		}
+		if tuple.hiddenResp.RespType != libkb.MerkleHiddenResponseTypeABSENCEPROOF && tuple.hiddenResp.RespType != libkb.MerkleHiddenResponseTypeFLAGOFF {
```
Comment (maxtaco) [inline go/teams/audit.go]: are we actually using these yet?

Comment (AMarcedone) [reply go/teams/audit.go]: we are using absence proof (they can either be absence proofs or inclusion proofs for empty leaves). 
However, the whole pr is meant to be merged after we start putting the hidden chain root in the main tree in prod.

**Comment by maxtaco** [comment]
I'd make sure CI works, it seems like it needs a server change

**Pull request merged**